### PR TITLE
[CARBONDATA-1108][CARBONDATA-1112] Supported IUD for vector reader in Spark 2.1

### PR DESCRIPTION
**Pull request by ravipesala**
Current vector code does not support delete operation, this PR adds the delete functionality.
And also fixes the nullpointer exception while initializing IUD analysis rule.

### CODE REVIEW

File: core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedVectorResultCollector.java [hunk 1]
```diff
@@ -144,6 +144,8 @@ protected void prepareDimensionAndMeasureColumnVectors() {
         return;
       }
       fillColumnVectorDetails(columnarBatch, rowCounter, requiredRows);
+      scannedResult
+          .markFilteredRows(columnarBatch, rowCounter, requiredRows, columnarBatch.getRowCounter());
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedVectorResultCollector.java]: please break the line after `(`

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2131/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2134/

File: core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedVectorResultCollector.java (hunk 1 above)
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedVectorResultCollector.java]: ok

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2135/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2136/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2139/

**Comment by chenliang613** [comment]
LGTM. verified at my local machine.

**Pull request closed without merging**